### DISCUSSION TITLE
How to add attributes to subgraphs

### DISCUSSION THREAD

**Discussion by zhinanchezhoubo** [pyg-team/pytorch_geometric#4696] [Q&A]
for example, I want to add properties to each subgraph,
```python
import torch
from torch_geometric.datasets import TUDataset

dataset = TUDataset(root='data/TUDataset', name='MUTAG')
for data in dataset:
  data.node_label=torch.tensor(1)
print(dataset[0].node_label)
```

But the following error is reported：
KeyError: 'node_label'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
[/usr/local/lib/python3.7/dist-packages/torch_geometric/data/storage.py](https://localhost:8080/#) in __getattr__(self, key)
     62         except KeyError:
     63             raise AttributeError(
---> 64                 f"'{self.__class__.__name__}' object has no attribute '{key}'")
     65 
     66     def __setattr__(self, key: str, value: Any):

AttributeError: 'GlobalStorage' object has no attribute 'node_label'

**Comment by rusty1s** [comment]
Any `dataset` access will return a new shallow-copy of the underlying data. To modify, you should make use of `transforms`:
```python
def my_transform(data):
    data.node_label = torch.tensor(1)
    return data

dataset = TUDataset(..., transform=my_transform)
print(dataset[0].node_label)
```

**Reply by zhinanchezhoubo** [reply]
thank you very much.
